Added Impact to the use cases to justify requirements.

sridharangirish91 opened the pull request:
Created a section to define all requirements and link them in the use cases for easier read.
Reworded a few things

sridharangirish91 committed:
justifying requirements by including impact of the requirements when they are met

sridharangirish91 committed:
Remove priority inversion and example to make it more generic

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -349,8 +342,73 @@ network elements may drop loss tolerant packets during reactive
+  * QUIC CID mapping
+  * ICMP messages
+
+## System-Considerations
```
```suggestion
## System Considerations
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -349,8 +342,73 @@ network elements may drop loss tolerant packets during reactive
+## System-Considerations
+
+REQ-PRIVACY-ADDITIONAL:
+: An on-path observer obtains no additional information about the IP
```
```suggestion
: An on-path observer obtains (or gleans) no additional information about the IP
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -349,8 +342,73 @@ network elements may drop loss tolerant packets during reactive
+packet.
+
+REQ-SIGNALING-AVOIDANCE:
+: Leveraging previous experience {{?RFC9049}}, the folliwing is not
```
```suggestion
: Leveraging previous experience {{?RFC9049}}, the following is not
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -349,8 +342,73 @@ network elements may drop loss tolerant packets during reactive
+
+  * Reveal the application identity.
+  * Expose the application cause (or 'reason') to signal metadata.
+  * Reveal server identity.
```
```suggestion
  * Reveal the server identity.
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -349,8 +342,73 @@ network elements may drop loss tolerant packets during reactive
+  * Inspect client-to-server encrypted payload by network elements.
+
+REQ-ISP-SCALE:
+: The metadata other state information that a wireless router has to
```
```suggestion
: The metadata and any other related state information that a network element (router) has to
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -349,8 +342,73 @@ network elements may drop loss tolerant packets during reactive
+
+REQ-ISP-SCALE:
+: The metadata other state information that a wireless router has to
+maintain for each additional media flow it handles should be very low
```
```suggestion
maintain for each additional flow it handles should be very low
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -349,8 +342,73 @@ network elements may drop loss tolerant packets during reactive
+: The network needs to ensure the signal is coming from the same
+user/client that is part of the 5-tuple flow.  This is to ensure no
+other application influences the priority of another application's
+flow.
```
```suggestion
flow from within the same host.
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -363,27 +421,48 @@ Examples:
-The client may indicate this preference to the network by selecting the importance value of audio packets as the highest priority.<br>
-<br>
-Requirement: REQ-PAYLOAD-CLIENT-DECIDES.
+The client may indicate this preference to the network by selecting
```
```suggestion
The client may indicate this preference to a network by selecting
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -363,27 +421,48 @@ Examples:
+    REQ-PAYLOAD-CLIENT-DECIDES (defined in {{x-req-def}}).
+
+    Impact: With the above requirement met, audio clarity could be maintained
+  better in challenging network and during reactive events ensuring better user experience.
```
```suggestion
  better in resource-constrained networks and during reactive events ensuring better user experience.
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -363,27 +421,48 @@ Examples:
-<br>
-Requirement: REQ-PAYLOAD-CLIENT-DECIDES.
+Different clients have different priorities for different types of traffic. This would result
+in change in priorities for the same type of traffic a single server sends, based on the user/client.
```
```suggestion
in change in priorities for the same type of traffic that a single server sends, based on the user/client.
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -363,27 +421,48 @@ Examples:
+
+    Impact: With the above requirement met, each client/user preferences are
+    prioritized accordingly while maintaining scalability on the server, since
+    the metadata the server sends still remains the same for all the connections.
```
```suggestion
    the metadata that the server sends still remains the same for all the connections.
```

boucadair commented on draft-kwbdgrr-tsvwg-net-collab-rqmts.md:
```diff
@@ -363,27 +421,48 @@ Examples:
+
+    Requirement: REQ-PACKET-PRIORITY.
+
+    Impact: By identifying packets that can be dropped, congestion can be reduced leading
```
```suggestion
    Impact: By identifying packets that tolerate being dropped, congestion can be reduced leading
```

sridharangirish91 committed:
Apply suggestions from code review

Co-authored-by: Med <mohamed.boucadair@orange.com>

boucadair commented:
Thanks.

For "REQ-METADATA-ACCURACY", please see my comment to Dan at #86.

John-Kaippallimalil committed:
Update draft-kwbdgrr-tsvwg-net-collab-rqmts.md

MEDIA-AV-SEPARATE and REQ-PAYLOAD-CLIENT-DECIDES that are in the Appendix are now referred to here in the main body. And it makes it normative. (unlike the Network Attachment sections in Appendix that are informative).

This revision indicates that the above requirements are "enhanced" (and not the basic set of requirements )

John-Kaippallimalil committed:
Update draft-kwbdgrr-tsvwg-net-collab-rqmts.md

Added explanation about REQ-MEDIA-KEYFRAME (ln 455) and REQ-PAYLOAD-CLIENT-DECIDES (ln 511) as enhanced requirements

John-Kaippallimalil commented:
The main body of the draft now refers to 3 requirements (REQ-MEDIA-AV-SEPARATE; REQ-PAYLOAD-CLIENT-DECIDES, REQ-MEDIA-KEYFRAME) that are in the Appendix. This blurs the line on what sections in the Appendix are normative and informative. As a partial remedy, I have made 2 updates  (see my commits above) to indicate that these requirements are "enhanced".

sridharangirish91 commented:
> The main body of the draft now refers to 3 requirements (REQ-MEDIA-AV-SEPARATE; REQ-PAYLOAD-CLIENT-DECIDES, REQ-MEDIA-KEYFRAME) that are in the Appendix. This blurs the line on what sections in the Appendix are normative and informative. As a partial remedy, I have made 2 updates (see my commits above) to indicate that these requirements are "enhanced".

Hi John, thanks for the text. I agree with REQ-MEDIA-AV-SEPARATE and REQ-MEDIA-KEYFRAME but I doubt if REQ-PAYLOAD-CLIENT-DECIDES causes any issue with traffic analysis. I have moved the statements you have made into the requirements definition itself to keep the main body short and succinct while keeping the remedy in place as well.

sridharangirish91 committed:
moved things to keep main body short while preserving the context

sridharangirish91 committed:
nit

John-Kaippallimalil commented:
Hi Sridhar, Re: traffic analysis comment may apply for QUIC - where a the sender may schedule packets in any order (RR, priority, deadline, or even mix multiple media types in different FRAMES of the same packet). Plus there is the option to add chaff packets or stuff packets to not be able to detect based on size/frequency. This enhanced metadata should obviously not be used where effort is being taken to mask it from middleboxes in the network. On the other hand RTP is different (except if it is RoQ).

Having this kind of dependency between Appendix and main body is not ideal in my view. The gist of the impact/effect on other parts of the main body should be described there, as otherwise its hard for a reader to figure out what the impact would be to realize that requirement. Its fine if the preference is to to keep more of the details in the Appendix.

danwing commented:
> Hi Sridhar, Re: traffic analysis comment may apply for QUIC - where a the sender may schedule packets in any order (RR, priority, deadline, or even mix multiple media types in different FRAMES of the same packet). Plus there is the option to add chaff packets or stuff packets to not be able to detect based on size/frequency. This enhanced metadata should obviously not be used where effort is being taken to mask it from middleboxes in the network. On the other hand RTP is different (except if it is RoQ).

Seems best to discuss once in Security Considerations (or Privacy Considerations) section, which we already discuss at https://github.com/SpencerDawkins/net-collab-rqmts/blob/main/draft-kwbdgrr-tsvwg-net-collab-rqmts.md#privacy-considerations-privacy

sridharangirish91 commented:
> The main body of the draft now refers to 3 requirements (REQ-MEDIA-AV-SEPARATE; REQ-PAYLOAD-CLIENT-DECIDES, REQ-MEDIA-KEYFRAME) that are in the Appendix. This blurs the line on what sections in the Appendix are normative and informative. As a partial remedy, I have made 2 updates (see my commits above) to indicate that these requirements are "enhanced".

If that's the case, then we can mention media streaming in extended use case as well, where we go little deeper into the use case and explain the extended requirements there while keeping the basic ones in the main section of the draft.

We can combine 1 and 3 in media streaming as just a way to prioritize different traffic differently and least priority ones can tolerate loss. We can then move 2 and add a use case where we can get into audio video separately as well as key frame there.
Alternatively, we could move CLIENT-PAYLOAD-DECIDES to the main doc and keep 2 there. Given AV and KeyFrame requirements are not required for Client-payload-decides and doesn't have any traffic analysis complications, it could be a valuable client-to-network signal in the main doc too.

sridharangirish91 committed:
addressing review comments and moving things around

sridharangirish91 commented:
@John-Kaippallimalil @danwing - Updated the text in the latest commit to keep extended use cases and requirements contained in the appendix.

John-Kaippallimalil reviewed (approved):
HI Sridhar,
This revision looks great. Thanks for the updates.

John-Kaippallimalil reviewed (approved):
Looks good.
